### PR TITLE
add no lint error checks for signalfx exporter package

### PR DESCRIPTION
**Pull request by DiptoChakrabarty**
**Description:** <Describe what has changed.>

This adds error checks for the signalf exporter package

**Link to tracking Issue:** #9749 

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: exporter/signalfxexporter/eventclient.go [hunk 1]
```diff
@@ -94,7 +94,10 @@ func (s *sfxEventClient) pushLogsData(ctx context.Context, ld plog.Logs) (int, e
 	}
 
 	defer func() {
-		io.Copy(ioutil.Discard, resp.Body)
+		_, err := io.Copy(ioutil.Discard, resp.Body)
+		if err != nil {
+			log.Fatalln(err)
```
Comment (bogdandrutu) [inline exporter/signalfxexporter/eventclient.go:99]: use `s.logger`, and make it an warning not a fatal level.

File: CHANGELOG.md [hunk 2]
```diff
@@ -69,6 +69,7 @@
 
 ### 🧰 Bug fixes 🧰
 
+- `signalfexporter`: add no lint error checks for signalf exporter package (#10258)
```
Comment (dmitryax) [inline CHANGELOG.md:72]:
```suggestion
- `signalfxexporter`: add no lint error checks for signalfx exporter (#10258)
```

File: exporter/signalfxexporter/eventclient.go (hunk 1 above)
Comment (DiptoChakrabarty) [reply exporter/signalfxexporter/eventclient.go]: done

File: exporter/signalfxexporter/exporter_test.go [hunk 3]
```diff
@@ -421,7 +420,12 @@ func TestConsumeMetricsWithAccessTokenPassthrough(t *testing.T) {
 			sfxExp, err := NewFactory().CreateMetricsExporter(context.Background(), componenttest.NewNopExporterCreateSettings(), cfg)
 			require.NoError(t, err)
 			require.NoError(t, sfxExp.Start(context.Background(), componenttest.NewNopHost()))
-			defer sfxExp.Shutdown(context.Background())
+			defer func() {
+				err = sfxExp.Shutdown(context.Background())
+				if err != nil {
+					log.Fatalln(err)
+				}
+			}()
```
[the remark this answers is not in the document]
Comment (DiptoChakrabarty) [reply exporter/signalfxexporter/exporter_test.go]: done

**Comment by DiptoChakrabarty** [comment]
Selection_004
How to resolve this error , can the tests be re run

**Comment by DiptoChakrabarty** [comment]
can the tests be re run

File: CHANGELOG.md [hunk 4]
```diff
@@ -69,6 +69,7 @@
 
 ### 🧰 Bug fixes 🧰
 
+- `signalfxexporter`: add no lint error checks for signalfx exporter package (#10258)
```
Comment (dmitryax) [inline CHANGELOG.md:72]: Please remove the changelog entry. This is not a user/developer facing change

**Comment by dmitryax** [comment]
@DiptoChakrabarty please fix remaining issues

**Comment by dmitryax** [comment]
Hey @DiptoChakrabarty , there are still linter issue. Please let us know if you can fix them.

```
golangci-lint run --allow-parallel-runners
Error: dpclient.go:140:9: Error return value of `io.Copy` is not checked (errcheck)
	io.Copy(ioutil.Discard, resp.Body)
	       ^
Error: exporter_test.go:676:25: Error return value of `sfxExp.Shutdown` is not checked (errcheck)
			defer sfxExp.Shutdown(context.Background())
			                     ^
Error: factory_test.go:650:20: Error return value is not checked (errcheck)
	setDefaultExcludes(cfg)
	                  ^
Error: eventclient.go:96:6: shadow: declaration of "err" shadows declaration at line 64 (govet)
		_, err := io.Copy(ioutil.Discard, resp.Body)
		   ^
```

**Review by bogdandrutu — APPROVED** [review]

File: exporter/signalfxexporter/eventclient.go [hunk 5]
```diff
@@ -94,7 +93,10 @@ func (s *sfxEventClient) pushLogsData(ctx context.Context, ld plog.Logs) (int, e
 	}
 
 	defer func() {
-		io.Copy(ioutil.Discard, resp.Body)
+		_, err := io.Copy(ioutil.Discard, resp.Body)
+		if err != nil {
+			s.logger.Warn(err.Error())
+		}
```
Comment (bogdandrutu) [inline exporter/signalfxexporter/eventclient.go:99]: ignore the error for the moment, don't log it.

**Comment by atoulme** [comment]
Closing, the code changes are no longer necessary. Please comment or reopen if changes are still needed.

**Pull request closed without merging**